crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2903,24 +3021,24 @@ pub async fn filter_payment_methods(
                         })
                         .unwrap_or(true);
 
-                    let connector = connector.clone();
+                    let context = AnalysisContext::from_dir_values(context_values.clone());
 
-                    let response_pm_type = ResponsePaymentMethodIntermediate::new(
-                        payment_method_object,
-                        connector,
-                        payment_method,
+                    let result = graph.key_value_analysis(
+                        pm_dir_value.clone(),
+                        &context,
+                        &mut cgraph::Memoization::new(),
+                        &mut cgraph::CycleCheck::new(),
+                        None,
                     );
-
-                    if filter
-                        && filter2
-                        && filter3
-                        && filter4
-                        && filter5
-                        && filter6
-                        && filter7
-                        && filter8
-                        && filter9
+                    if filter_pm_based_on_allowed_types
+                        && filter_incorrect_client_secret
+                        && result.is_ok()
```

Comment:
We need to check if the result is `true`. Also, log the error if we get one.
```suggestion
                        && matches!(result, Ok(true))
```

Reply:
`key_value_analysis` returns `Result<(), E>`, So matching on `Ok(true)` won't be possible we can anyways match on `Ok(())`.